syntax = "proto3";

package suggestions;

// Service definition
service SuggestionsService {
    rpc InitOrder (InitOrderRequest) returns (InitOrderConfirmationResponse);
    rpc GetBookSuggestions (SuggestionsRequest) returns (SuggestionsResponse);
}

// Request message (we will use the items ordered to suggest similar books)
message SuggestionsRequest {
    string order_id = 1;
    repeated int32 vc = 2;
}

// Item message (each item in the order)
message Item {
    string name = 1;
    int32 quantity = 2;
}

// Response message (list of suggested books)
message SuggestionsResponse {
    repeated Book books = 1;
    repeated int32 vc = 2;
}

// Book message (represents a suggested book)
message Book {
    string bookId = 1;
    string title = 2;
    string author = 3;
}

message OrderData {
    repeated Item items = 1;
}

message InitOrderRequest {
    string order_id = 1;
    OrderData order_data = 2;
}

message InitOrderConfirmationResponse {
    bool isCreated = 1;
}
